### ISSUE TITLE
Gem unable to read PDF contents. Error: PDF::Reader::EncryptedPDFError Unsupported encryption style

### ISSUE THREAD

**Issue by luigilake** [yob/pdf-reader#262]
Hey guys! I’m trying to open a pdf file and go through its contents within a Ruby on Rails app. Instructions seems straightforward, but I hit an error that I can’t seem to figure out. It looks like the PDF I’m trying to access is encrypted. This is my code:

```require 'open-uri'
require 'rubygems'
require 'pdf/reader'

io = open(https://www.lightingglobal.org/wp-content/uploads/2013/12/LG_SSS-dl-S20_AR.pdf)
reader = PDF::Reader.new(io)
puts reader.info
```
and the error is:
```PDF::Reader::EncryptedPDFError
Unsupported encryption style
```
Does anyone know how to access encrypted pdfs?

in the source code for the gem, `StandardSecurityHandler` is checking this:

```def self.supports?(encrypt)
   return false if encrypt.nil?
   binding.pry
   filter = encrypt.fetch(:Filter, :Standard)
   version = encrypt.fetch(:V, 0)
   algorithm = encrypt.fetch(:CF, {}).fetch(encrypt[:StmF], {}).fetch(:CFM, nil)
   filter == :Standard && (version <= 3 || (version == 4 && algorithm != :AESV2))
end
```

It looks like it might need something called Advanced Encryption Standard 2 to be decrypted, but I'm unfamiliar with this part of it. 

These pdfs are quite crucial to the project, so any help with this would be appreciated! Thank you!

**Comment by yob** [comment]
Hi,

The master branch of this repo has support for additional encryption types, and I've confirmed it can read the test file you linked to. Are you able to load the gem via a git reference in your Gemfile?

**Comment by JackSoby** [comment]
Hi, 

 I am trying to scrape the same PDF as Luigi. I was looking at your example code of a grabbing info from a encrypted pdf. I have something like this  ``` reader =PDF::Reader.new(https://www.lightingglobal.org/wp-content/uploads/2013/12/LG_SSS-dl-S20_AR.pdf , :password => "?")```  How am I supposed to tell what to put for a password?

**Comment by yob** [comment]
> How am I supposed to tell what to put for a password?

Does a blank password work? If not then I'm afraid I can't help - you'll have yo ask the person who encrypted the file

**Comment by JackSoby** [comment]
Would I just put an empty string in as the password? If so an empty string does not work. As you said before you have confirmed it can read the test file I linked to. How did you get that link to work?

**Comment by yob** [comment]
Using the current master branch of this repo, the following works for me:

```ruby
require 'pdf-reader'

reader = PDF::Reader.new('LG_SSS-dl-S20_AR.pdf', password: "")
puts reader.page_count
```

However, an empty string password is the default so the following works too:

```ruby
require 'pdf-reader'

reader = PDF::Reader.new('LG_SSS-dl-S20_AR.pdf')
puts reader.page_count
```

**Comment by JackSoby** [comment]
First off I would just like to say I really appreciate all the help, you have been super helpful. I know your time is valuable and I am very  grateful for our feedback.  

So I originally had this 

``` 
 spec_pdf = https://www.lightingglobal.org/wp-content/uploads/2013/12/LG_SSS-dl-S20_AR.pdf

      io = open(spec_pdf)

      reader = PDF::Reader.new(io, :password => '')
```

When I run this I get the error ```Unsupported encryption style``` 


Instead i tried running 

```reader = PDF::Reader.new('LG_SSS-dl-S20_AR.pdf')```

And I got the error 

```input must be an IO-like object or a filename```

Which I understand means the file does not exist. So I tried running this instead 

```  spec_pdf ="LG_SSS-dl-S20_AR.pdf"
      io = open(spec_pdf)
reader = PDF::Reader.new(io)
```

And this tells me no such file or directory exists. Not quite sure what is different from yours or mine.

**Comment by yob** [comment]
Hi @JackSoby - this is a ruby question rather than something pdf-reader specific.

For the `open()` method to accept a URL you need to first require 'open-uri', like this:

```ruby
require 'pdf-reader'
require 'open-uri'

spec_pdf = "https://www.lightingglobal.org/wp-content/uploads/2013/12/LG_SSS-dl-S20_AR.pdf"
reader = PDF::Reader.new(open(spec_pdf))
puts reader.page_count
```

Alternatively, you could manually download the PDF to the same directory as your ruby script and then the code I posted earlier will work:

```ruby
require 'pdf-reader'

reader = PDF::Reader.new("LG_SSS-dl-S20_AR.pdf")
puts reader.page_count
```

**Comment by JackSoby** [comment]
So this code is still giving me the ```Unsupported encryption style``` 
```
  require 'open-uri'
  spec_pdf = "https://www.lightingglobal.org/wp-content/uploads/2013/12/LG_SSS-dl-S20_AR.pdf"
  reader = PDF::Reader.new(open(spec_pdf))
  puts reader.page_count
```

Am I too assume that the only way for me to scrape this pdf is to download it onto my local machine and put it in the same directory as my project?

**Comment by yob** [comment]
> So this code is still giving me the Unsupported encryption style

The latest released version of pdf-reader unfortunately doesn't support the encryption used in that file.

To read it, you'll need to load pdf-reader via git in your Gemfile. Sorry!

**Comment by JackSoby** [comment]
that worked! thank you!!!!!!

**Comment by rubish** [comment]
@yob I am also using the gem via git to get support for additional encryption types which is available in released gem. Any plans to release a gem for the latest master?

**Comment by yob** [comment]
@rubish 

>  Any plans to release a gem for the latest master?

Thanks for the prompt - I've just released v2.1.0

**Comment by rubish** [comment]
Awesome :)
